pyccel/parser/scope.py

```diff
@@ -874,3 +910,31 @@ def collect_tuple_element(self, tuple_elem):
                 return result
         else:
             return tuple_elem
+
+    def collect_all_tuple_elements(self, tuple_var):
+        """
+        Create a tuple of variables from a variable representing an inhomogeneous object.
+
+        Create a tuple of variables that can be printed in a low-level language. An
+        inhomogeneous object cannot be represented as is in a low-level language so
+        it must be unpacked into a PythonTuple. This function is recursive so that
+        variables with a type such as `tuple[tuple[int,bool],float]` generate
+        `PythonTuple(PythonTuple(var_0_0, var_0_1), var_1)`.
+
+        Parameters
+        ----------
+        tuple_var : Variable
+            A variable which may or may not be an inhomogeneous tuple.
+
+        Returns
+        -------
+        Variable | PythonTuple
+            An object containing only variables that can be printed in a low-level language.
```

Comment:
A `list` may also be returned by this function

Reply:
Fixed 404f55d9c

Reply:
Fixed c0ebb9973